Modules/Common/AvalancheCrossChainLikeMainModule.php

```diff
@@ -0,0 +1,291 @@
+                    'i',
+                    $atomic_events,
+                    $currencies_used
+                );
+            }
+
+            if (array_key_exists('outputs', $atomic_transaction)) {
+                $this->parse_transaction_legs(
+                    $atomic_transaction['outputs'],
+                    $atomic_transaction['hash'],
+                    $ittr,
+                    4,
+                    "",
+                    null,
+                    $atomic_events,
+                    $currencies_used
+                );
+            }
+
+            if (array_key_exists('exports', $atomic_transaction)) {
+                $this->parse_transaction_legs(
+                    $atomic_transaction['exports'],
+                    $atomic_transaction['hash'],
+                    $ittr,
+                    5,
+                    "",
+                    'x',
+                    $atomic_events,
+                    $currencies_used
+                );
+
+            }
+
+            $ittr++;
+        }
+
+        return [$atomic_events, $currencies_used];
+    }
+
+    private function process_events($block_id, $events, $block_time)
+    {
+        $hmr_time = date('Y-m-d H:i:s', to_int64_from_0xhex($block_time));
+        foreach ($events as &$event) {
+            $event['block'] = $block_id;
+            $event['time'] = $hmr_time;
+        }
+
+        // Re-sort
+        usort($events, function($a, $b)
+        {
+            return
+                [$a['sort_in_block'], $a['sort_in_transaction'],]
+                <=>
+                [$b['sort_in_block'], $b['sort_in_transaction'],];
+        });
+
+        $sort_key = 0;
+
+        foreach ($events as &$event) {
+            $event['sort_key'] = $sort_key;
+            $sort_key++;
+
+            unset($event['sort_in_block']);
+            unset($event['sort_in_transaction']);
+        }
+
+        $this->set_return_events($events);
+    }
+
+    private function fetch_currency_description($currency)
+    {
+        return requester_single($this->asset_info_handle, params: [
+            'method' => 'avm.getAssetDescription',
+            'params' => [
+                'assetID' => $currency
+            ],
+            'id' => 0,
+            'jsonrpc' => '2.0',
+        ],
+        result_in: 'result', timeout: $this->timeout);
+    }
+
+    private function process_currencies($currencies_used)
+    {
+        $currencies = [];
+
+        foreach ($currencies_used as $id => $_) {
+            if ($id === $this->main_token_descr['id']) {
+                $currencies[] = [
+                    'id'       => $this->main_token_descr['id'],
+                    'name'     => $this->main_token_descr['name'],
+                    'symbol'   => $this->main_token_descr['symbol'],
+                    'decimals' => $this->main_token_descr['decimals']
+                ];
+            } else {
+                $cur_data = fetch_currency_description($id);
```

Comment:
This function is not defined. Probably you mean `$this->fetch_currency_description($id)`, but it means that this code has never been used/tested.

Reply:
It was indeed meant to be `$this->`. This branch has not been tested, for non-avax crosschain transactions are "still under development", although it has been a good 3 months since then, I better double-check if there has been any updates